src/v/cloud_storage/remote_segment.h

```diff
@@ -119,46 +119,23 @@ class remote_segment final {
       _wait_list;
 };
 
-/// Log reader config for shadow-indexing
+class remote_segment_batch_consumer;
+
+/// The segment reader that can be used to fetch data from cloud storage
 ///
-/// The difference between storage::log_reader_config and this
-/// object is that storage::log_reader_config stores redpanda
-/// offsets in start_offset and max_offset fields. The
-/// cloud_storage::log_reader_config stores kafka offsets in these
-/// fields. It also adds extra field 'start_offset_redpanda' which
-/// always contain the same offset as 'start_offset' but translated
-/// from kafka to redpanda.
+/// The reader invokes 'data_stream' method of the 'remote_segment'
+/// which returns hydrated segment from disk.
 ///
 /// The problem here is that shadow-indexing operates on sparse data.
 /// It can't translate every offset. Only the base offsets of uploaded
 /// segment. But it can also translate offsets as it scans the segment.
 /// But this is all done internally, so caller have to proviede kafka
 /// offsets. Mechanisms which require redpanda offset can use
-/// 'start_offset_redpanda' field. It's guaranteed to point to the same
+/// '_cur_rp_offset' field. It's guaranteed to point to the same
 /// record batch but the offset is not translated back to kafka. This is
 /// useful since we can, for instance, compare it to committed_offset of
 /// the uploaded segment to know that we scanned the whole segment.
-struct log_reader_config : public storage::log_reader_config {
```

Comment:
nit: maybe instead of removing the struct you can replace it with `using log_reader_config = storage::log_reader_config` so the code change will be more local